### PR TITLE
Added XFAIL to methods

### PR DESCRIPTION
**Pull request by cekbote**
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by sylee957** [comment]
Release notes should not be added for features that are invisible for users.

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> Release notes should not be added for features that are invisible for users.

Thank you. Will make the necessary change.

File: sympy/printing/tests/test_tensorflow.py [hunk 1]
```diff
@@ -129,6 +129,7 @@ def test_tensorflow_printing():
             " [tensorflow.math.exp(z), -t]])"
 
 
+@XFAIL
 def test_tensorflow_math():
```
Comment (oscarbenjamin) [inline sympy/printing/tests/test_tensorflow.py]: Later on someone might try to remove the XFAIL here if they can see the test passes. There should be a comment indicating why it is marked as XFAIL even though it usually passes. The comment should have a link to the issue.

**Comment by oscarbenjamin** [comment]
The OP should have a reference to the issue that you are addressing. It should not say "fixes" though as this does not fix the issue (it just makes it less annoying).

File: sympy/printing/tests/test_tensorflow.py (hunk 1 above)
Comment (Chanakya-Ekbote) [reply sympy/printing/tests/test_tensorflow.py]: Ok sure. I'll add that. Thank you.

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> The OP should have a reference to the issue that you are addressing. It should not say "fixes" though as this does not fix the issue (it just makes it less annoying).

Sure. Apologies for the inconvenience caused.

File: sympy/printing/tests/test_tensorflow.py (hunk 1 above)
Comment (smichr) [reply sympy/printing/tests/test_tensorflow.py]: Why are we testing random numbers? If we are just trying to see if the tensor flow result is the same as the result when evaluating the expr at the same value then why not just choose a non-pathological value like `-pi` to test for each of them?

**Comment by sylee957** [comment]
I have just extended what the original tests were organized before, after fixing some errors
But it looks like tensorflow precision is poor for matrix inverse or acos, and I’d try to revert everything with selective test cases.

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> I have just extended what the original tests were organized before, after fixing some errors
> But it looks like tensorflow precision is poor for matrix inverse or acos, and I’d try to revert everything with selective test cases.

So should I create dedicated tests for matrix inversion or for all the problematic ones?

**Comment by sylee957** [comment]
Yes, it can be better to make matrix inversion test non-random, and also ‘acos’

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> Yes, it can be better to make matrix inversion test non-random, and also ‘acos’

So what I noticed in acos was that if the random value gets too high (above a threshold of 0.995) tensorflow and sympy give different values. Hence the max limit I've added for acos is 0.99. I have attached the photo below which shows the discrepancy.
Capture

I have yet to figure out the issue for the Inverse. It may be the same as before where the limits may have to be changed.

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> > Yes, it can be better to make matrix inversion test non-random, and also ‘acos’
> 
> So what I noticed in acos was that if the random value gets too high (above a threshold of 0.995) tensorflow and sympy give different values. Hence the max limit I've added for acos is 0.99. I have attached the photo below which shows the discrepancy.
> Capture
> 
> I have yet to figure out the issue for the Inverse. It may be the same as before where the limits may have to be changed.

If I have to write custom tests for the Inverse method, how many should I write?

**Comment by sylee957** [comment]
Only one can be sufficient

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> Only one can be sufficient

I have added a test. Could you please le me know if its ready to be merged?

File: sympy/printing/tests/test_tensorflow.py [hunk 2]
```diff
@@ -129,6 +161,8 @@ def test_tensorflow_printing():
             " [tensorflow.math.exp(z), -t]])"
 
 
+# The XFAIL is added because of the PR: https://github.com/sympy/sympy/issues/18090
+@XFAIL
```
Comment (oscarbenjamin) [inline sympy/printing/tests/test_tensorflow.py]: I think it would be better to write
```
# This (random) test is XFAIL because it fails occasionally
# See https://github.com/sympy/sympy/issues/18090
```

**Comment by oscarbenjamin** [comment]
The OP here still does not reference the related issue

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> The OP here still does not reference the related issue

Apologies. Do you mean in the code or in this issue? If its in this issue, then I have addressed to a certain degree, issue #18090 

> 
> 
> The OP here still does not reference the related issue

Aplogies. If you mean in the code, I have added the changes you have mentioned above.
If you mean in this issue, then I have addressed issue #18090 to a certain extent.

**Comment by sylee957** [comment]
I think that the changes with the plot still need to be reverted.
I still worry that things can fail near `0.99`, why not set it like `0.95`?

**Comment by oscarbenjamin** [comment]
This is the OP: https://github.com/sympy/sympy/pull/18469#issue-367218195

**Comment by Chanakya-Ekbote** [comment]
I apologise. My computer got corrupted, and then while resetting my system, I accidentally deleted the remote branch also (while deleting a few other branches). Can I start a new PR? Apologies once again for the inconvenience caused.

**Comment by Chanakya-Ekbote** [comment]
I have reverted the changes in ```plot.py``` and have also changed the comments as well included the value of 0.95 instead of 0.99. Is there something else I should do or can this be merged?

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> I have reverted the changes in `plot.py` and have also changed the comments as well as included the value of 0.95 instead of 0.99 in `test_tensorflow.py`. Is there something else I should do or can this be merged?

@sylee957 @oscarbenjamin

File: sympy/printing/tests/test_tensorflow.py (hunk 2 above)
Comment (oscarbenjamin) [reply sympy/printing/tests/test_tensorflow.py]: I think it would be better to refer to the issue (#18090) than this PR (#18469)

File: sympy/printing/tests/test_tensorflow.py [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import random
-
```
Comment (oscarbenjamin) [inline sympy/printing/tests/test_tensorflow.py]: Why is this line deleted?

**Comment by oscarbenjamin** [comment]
Can you squash this into one commit? Lots of unrelated changes have happened in the previous commits.

The way to squash it is like this:
```
git checkout test # go to this branch
git checkout -b test_backup # make a backup branch!
git checkout test # back to this branch
git log # find the first commit that is not yours and copy the hash (e.g. 1234)
git reset 1234 # The hash you copied
git reset --hard # This deletes everything. Make sure you backed up!
git status
git diff
git commit -a # Make a new commit for the changes
git push --force # This pushes the squashed commits here in the PR.
```

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> Can you squash this into one commit? Lots of unrelated changes have happened in the previous commits.
> 
> The way to squash it is like this:
> 
> ```
> git checkout test # go to this branch
> git checkout -b test_backup # make a backup branch!
> git checkout test # back to this branch
> git log # find the first commit that is not yours and copy the hash (e.g. 1234)
> git reset 1234 # The hash you copied
> git reset --hard # This deletes everything. Make sure you backed up!
> git status
> git diff
> git commit -a # Make a new commit for the changes
> git push --force # This pushes the squashed commits here in the PR.
> ```

I apologise for these messy commits. I will make sure not make the same mistake again. I did try to do what you asked for, however, there is an issue. My computer got corrupted in the middle and while doing some cleaning I accidentally deleted my test remote. Although I did start a new branch called test, I have only the logs after Feb 1st. Hence is there some alternative? Apologies once again for this mess. The only code that has been changed is in ```test_tensorflow```. The rest all are just reverts (I was messing around with a few Sympy functions and then forgot about that).

**Comment by oscarbenjamin** [comment]
I have squashed the commit myself. If you try to push or pull this PR now you will have problems.

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> I have squashed the commit myself. If you try to push or pull this PR now you will have problems.

Alright. Thanks. I won't mess with it anymore. Apologies once again for the mess.

**Comment by oscarbenjamin** [comment]
Ok, thanks for this. I'll merge it now

**Comment by Chanakya-Ekbote** [comment]
> 
> 
> Ok, thanks for this. I'll merge it now

Thank you for all your help. It was fun contributing.
